### PR TITLE
Fix bugs in RemoteLinksTests

### PR DESCRIPTION
**Pull request by romanowski**
Addresses some of the comments from #13196

### CODE REVIEW

**Comment by griggt** [comment]
The changes in this PR currently look identical to #13201.  Did you mean for the changes here to be those in feeb1d75b5b99cda8d318a380bd9327ac8eb707f ?

**Comment by romanowski** [comment]
I somehow messed-up my branches. It should be good now, sorry!

File: scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala [hunk 1]
```diff
@@ -38,16 +38,17 @@ class RemoteLinksTest:
   def runTest =
     assertTrue(mtslAll.nonEmpty)
     val mtsl = randomGenerator.shuffle(mtslAll).take(80) // take 80 random entries
-    val pageToMtsl: Map[String, List[(String, String)]] = mtsl.groupMap(_._2.split("#L").head)(v => (v._1, v._2.split("#L").last))
-    pageToMtsl.foreach { case (link, members) =>
+    val pageToMtsl: Map[String, List[(String, String)]] =
```
Comment (dwijnand) [inline scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala:41]:
```suggestion
    val pageToMtsl: Map[String, List[(String, Int)]] =
```

... if I eyeball correctly

File: scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala [hunk 2]
```diff
@@ -38,16 +38,17 @@ class RemoteLinksTest:
...
+    pageToMtsl.toSeq.sortBy(_._1).foreach { case (link, members) =>
       try
         val doc = getDocumentFromUrl(link)
         println(s"Checking $link")
         members.foreach { case (member, expectedLine) =>
           if !member.startsWith("given_") then // TODO: handle synthetic givens, for now we disable them from testing
             val toLine = expectedLine + 3
             val memberToMatch = member.replace("`", "")
-            val lineCorrectlyDefined = (expectedLine.toInt until toLine.toInt).exists{ line =>
+            val lineCorrectlyDefined = (expectedLine.toInt until toLine).exists{ line =>
```
Comment (griggt) [inline scaladoc/test-source-links/dotty/tools/scaladoc/source-links/RemoteLinksTest.scala:51]:
```suggestion
            val lineCorrectlyDefined = (expectedLine until toLine).exists{ line =>
```

in addition to the change suggested by @dwijnand

**Review by dwijnand — APPROVED** [review]

**Pull request merged**